src/string/mod.rs

```diff
@@ -28,20 +28,28 @@ pub use drain::Drain;
 #[derive(Debug)]
 pub enum FromUtf16Error {
     /// The capacity of the `String` is too small for the given operation.
-    Capacity,
+    Capacity(CapacityError),
     /// Error decoding UTF-16.
     DecodeUtf16Error(DecodeUtf16Error),
```

Comment:
Given `Capacity` is not called `CapacityError`, this variant should probably be named just `DecodeUtf16`.

Reply:
Done!  That's consistent with `std::io::ErrorKind` naming.  Makes sense to me.